### PR TITLE
PLAT-137544: Update Item styles

### PR DESCRIPTION
**Pull request by adrian-cocoara-lgp**
### Checklist

* [x] I have read and understand the [contribution guide](http://enactjs.com/docs/developer-guide/contributing/)
* [x] A [CHANGELOG entry](http://enactjs.com/docs/developer-guide/contributing/changelogs/) is included
* [x] At least one test case is included for this feature or bug fix
* [x] Documentation was added or is not needed
* [x] Documentation was verified or is not changed
* [x] UI test was passed or is not needed
* [x] Screenshot test was verified or is not needed
* [ ] This is an API breaking change

### Resolution
[//]: # (Does the code work as intended?)
[//]: # (What is the impact of this change and *why* was it made?)
Updated Item according to the new design requirements. 
Fixed label vertical alignment for all skins for labelAfter and labelBefore
Fixed label horizontal alignement when centered and labelAbove, labelBelow
@agate-item-inline-content-max-width was set to 420px because of the bug described here: https://github.com/enactjs/agate/pull/547#r596154285
A MarqueeDecorator was added in order to overwrite Marquee's text alignment when centered is true. The label was not horizontally centered  when centered and labelAbove, labelBelow
Adding MarqueeDecorator directly to ItemBase was not possible because MarqueeController was already used and, if both are added, an error was thrown "cannot use a marquee inside another marquee"

### Links
[//]: # (Related issues, references)
PLAT-137544

### Comments

Enact-DCO-1.0-Signed-off-by: Adrian Cocoara adrian.cocoara@lgepartner.com

### CODE REVIEW

**Comment by daniel-stoian-lgp** [comment]
Item without label height should have height 100px. 

Item selected icon position is not correct. Maybe you have to remove Item padding

image


Opacity of border and text-color are not the same as in UI spec

image

**Comment by daniel-stoian-lgp** [comment]
adding this in variables-base affected all skins:
image
Please check

File: Item/Item.module.less [hunk 1]
```diff
@@ -134,10 +143,14 @@
 			}
 		}
 
-		&.large { /* Needed to prevent global class being added in the DOM */ }
+		&.large {
+			height: @agate-item-large-height;
+			line-height: @agate-item-line-height-large;
```
Comment (daniel-stoian-lgp) [inline Item/Item.module.less:148]: if you add line-height here, move the line-height used at line 49 here

File: styles/colors-carbon.less [hunk 2]
```diff
@@ -98,6 +98,7 @@
 @agate-item-label-color: @agate-accent;
 @agate-item-focus-bg-color: @agate-spotlight-bg-color;
 @agate-item-focus-color: @agate-spotlight-color;
+@agate-item-label-focus-selected-color: @agate-spotlight-color;
```
Comment (daniel-stoian-lgp) [inline styles/colors-carbon.less:101]: this variable is not declare in colors-base.less. Can you declare it there 
@agate-item-label-focus-selected-color: @agate-spotlight-color;
and remove it from the files of the skins which us the same value?

File: styles/variables-base.less [hunk 3]
```diff
@@ -322,8 +322,8 @@
 // ---------------------------------------
 @agate-indicator-align-self:    inherit;
 @agate-indicator-border-radius: inherit;
-@agate-indicator-display:       inherit;
-@agate-indicator-height:        inherit;
+@agate-indicator-display:       0;
+@agate-indicator-height:        0;
```
Comment (daniel-stoian-lgp) [inline styles/variables-base.less:326]: these 2 variables look to not be used in any component style. Can you check and remove them for all skins if necesary?

File: styles/variables-base.less [hunk 4]
```diff
@@ -396,28 +396,32 @@
 @agate-item-height:                   60px;
 @agate-item-height-large:             72px;
 @agate-item-indent:                   36px;
-@agate-item-inline-content-max-width: 240px;
+@agate-item-inline-content-max-width: 420px;
```
Comment (daniel-stoian-lgp) [inline styles/variables-base.less:399]: add this new value only for Silicon

File: styles/variables-base.less [hunk 5]
```diff
@@ -396,28 +396,32 @@
...
 @agate-item-line-height-large:        @agate-item-height-large - (@agate-item-vertical-padding * 2);
+@agate-item-max-width:                auto;
 @agate-item-side-outset:              @agate-component-spacing;
 @agate-item-slotafter-spacing:        @agate-component-spacing;
 @agate-item-slotbefore-spacing:       @agate-component-spacing;
 @agate-item-sub-text-size:            inherit;
 @agate-item-vertical-padding:         3px;
 @agate-item-small-font-size:          inherit;
-@agate-item-small-height:             54px;
+@agate-item-small-height:             99px;
```
Comment (daniel-stoian-lgp) [inline styles/variables-base.less:419]: add this new value only for Silicon

File: styles/variables-base.less [hunk 6]
```diff
@@ -396,28 +396,32 @@
 @agate-item-height:                   60px;
 @agate-item-height-large:             72px;
 @agate-item-indent:                   36px;
```
Comment (daniel-stoian-lgp) [inline styles/variables-base.less:398]: orphan variable. please remove it :)

File: styles/variables-base.less [hunk 7]
```diff
@@ -396,28 +396,32 @@
...
+@agate-item-labelafterbefore-label-max-width: 150px;
 @agate-item-label-font-size:          inherit;
 @agate-item-label-line-height:        inherit;
 @agate-item-label-max-width:          40%;
 @agate-item-label-padding:            10px;
+@agate-item-large-height:             120px;
 @agate-item-line-height:              @agate-item-height - (@agate-item-vertical-padding * 2);
 @agate-item-line-height-large:        @agate-item-height-large - (@agate-item-vertical-padding * 2);
+@agate-item-max-width:                auto;
 @agate-item-side-outset:              @agate-component-spacing;
```
Comment (daniel-stoian-lgp) [inline styles/variables-base.less:413]: orphan variable. please remove it :)

File: styles/variables-base.less [hunk 8]
```diff
@@ -396,28 +396,32 @@
...
 @agate-item-slotbefore-spacing:       @agate-component-spacing;
 @agate-item-sub-text-size:            inherit;
 @agate-item-vertical-padding:         3px;
 @agate-item-small-font-size:          inherit;
-@agate-item-small-height:             54px;
+@agate-item-small-height:             99px;
 @agate-item-small-itemcontent-line-height: inherit;
 @agate-item-small-line-height:        @agate-item-small-height - (@agate-item-vertical-padding * 2);
 @agate-item-small-label-font-size:    inherit;
 @agate-item-small-label-line-height:  inherit;
```
Comment (daniel-stoian-lgp) [inline styles/variables-base.less:423]: orphan variable. please remove it from base/gallium/silicon

File: Item/Item.module.less (hunk 1 above)
Comment (adrian-cocoara-lgp) [reply Item/Item.module.less:148]: This property is not used anymore.

File: styles/variables-base.less (hunk 4 above)
Comment (adrian-cocoara-lgp) [reply styles/variables-base.less:399]: This is needed for all skins because currently the layout is broken for inline Item
broken

File: styles/variables-base.less (hunk 3 above)
Comment (adrian-cocoara-lgp) [reply styles/variables-base.less:326]: They are used in Silicon skin to render the dot in selected state. They are not needed in other skins, so that's why I set them to `0`.

File: Item/Item.module.less [hunk 9]
```diff
@@ -156,38 +172,55 @@
...
 
-			.itemContent {
-				.label {
-					color: @agate-item-selected-label-color;
+			.enact-locale-rtl({
+				padding: @agate-item-selected-rtl-padding;
+
+				&::before {
+					margin-left: @agate-indicator-rtl-margin-left;
+					margin-right: @agate-indicator-rtl-margin-right;
```
Comment (daniel-stoian-lgp) [inline Item/Item.module.less:221]: this can be simplified. use padding-start-end() at line 204 and margin-start-end at line 213()

**Review by daniel-stoian-lgp — APPROVED** [review]

**Comment by jeonghee27** [comment]
Could you check the custom width and position of slotafter kitchen sink sampler on qa-sampler?
image

**Comment by jeonghee27** [comment]
Could you add size prop into qa-sampler in order to make easy to check?

File: Item/Item.module.less [hunk 10]
```diff
@@ -60,10 +61,12 @@
 
 				&.labelBefore .label {
 					.padding-start-end(0, @agate-item-label-padding);
+					align-self: center;
```
Comment (jeonghee27) [inline Item/Item.module.less:64]: Could you add a Centered case into Kitchen Sink?

File: Item/Item.module.less [hunk 11]
```diff
@@ -156,37 +177,48 @@
 			}
 		}
 
+		.disabled({
+			color: @agate-item-disabled-color;
+			border-color: @agate-item-disabled-border-color;
+
+			.focus({
+				opacity: @agate-item-disabled-opacity;
```
Comment (jeonghee27) [inline Item/Item.module.less:185]: I think the name is wrong.
@agate-item-focus-disabled-opacity?

File: styles/colors-silicon-day.less [hunk 12]
```diff
@@ -206,13 +207,17 @@
 // ---------------------------------------
 @agate-item-border-color:                 fade(@agate-black, 40%);
 @agate-item-color:                        @agate-text-color;
-@agate-item-label-color:                  @agate-noble-gray;
-@agate-item-focus-bg-color:               fade(@agate-black, 15%);
+@agate-item-disabled-border-color:        fade(@agate-foreground, 40%);
+@agate-item-disabled-color:               fade(@agate-foreground, 40%);
+@agate-item-disabled-opacity:             0.3;
```
Comment (jeonghee27) [inline styles/colors-silicon-day.less:212]: When I select disabled, the color is changed and opacity :0.6 also applied.

File: styles/colors-silicon-day.less [hunk 13]
```diff
@@ -206,13 +207,17 @@
 // ---------------------------------------
 @agate-item-border-color:                 fade(@agate-black, 40%);
 @agate-item-color:                        @agate-text-color;
-@agate-item-label-color:                  @agate-noble-gray;
-@agate-item-focus-bg-color:               fade(@agate-black, 15%);
+@agate-item-disabled-border-color:        fade(@agate-foreground, 40%);
+@agate-item-disabled-color:               fade(@agate-foreground, 40%);
```
Comment (jeonghee27) [inline styles/colors-silicon-day.less:211]: When I select disabled, the color is changed and opacity :0.6 also applied.

File: styles/colors-base.less [hunk 14]
```diff
@@ -234,14 +235,18 @@
...
+@agate-item-disabled-opacity:             @agate-disabled-opacity;
 @agate-item-label-color:                  @agate-item-color;
+@agate-item-label-focus-selected-color:   @agate-spotlight-color;
 @agate-item-focus-bg-color:               @agate-background-focus;
+@agate-item-focus-bg-image:               none;
 @agate-item-focus-border-color:           transparent;
 @agate-item-focus-color:                  @agate-foreground-focus;
 @agate-item-selected-bg-color:            @agate-item-bg-color;
 @agate-item-selected-border-color:        @agate-item-border-color;
 @agate-item-selected-color:               @agate-item-color;
```
Comment (jeonghee27) [inline styles/colors-base.less:249]: I think we should remove this. It depend on Gallium.
Should the font color change when selecting in Gallium?

when I set `disabled` and set `select`, the font color is changed. it is not needed on Silicon.

File: styles/variables-silicon.less [hunk 15]
```diff
@@ -116,6 +116,7 @@
 @agate-dropdown-button-height: 60px;
 @agate-dropdown-button-padding: 0 15px;
 @agate-dropdown-item-border-width: 0;
+@agate-dropdown-item-height: 42px;
```
Comment (jeonghee27) [inline styles/variables-silicon.less:119]: I think this should be 60px

**Comment by jeonghee27** [comment]
Please add a screenshot test with centered and slotbefore slotafter


Centered :
image

File: Item/Item.module.less (hunk 11 above)
Comment (adrian-cocoara-lgp) [reply Item/Item.module.less:185]: This had to be deleted. `opacity` was moved out of `focus` mixin, so the naming is ok

File: styles/colors-base.less (hunk 14 above)
Comment (adrian-cocoara-lgp) [reply styles/colors-base.less:249]: This seems to be solved by changing `@agate-item-disabled-color: fade(@agate-foreground, 40%);` to `@agate-item-disabled-color: @agate-foreground;`. Now the font color remains the same when disabled and selected

File: styles/variables-silicon.less (hunk 15 above)
Comment (adrian-cocoara-lgp) [reply styles/variables-silicon.less:119]: I wanted to keep the same height as it is now on develop (42px), but with the new specifications it is 60px. I changed it.

File: samples/sampler/stories/qa/Item.js [hunk 16]
```diff
@@ -91,6 +91,46 @@ storiesOf('Item', module)
...
+				<Item centered size="large">Large Item</Item>
+				<Item centered size="small">Small Item</Item>
+				<Item centered inline label="label below" labelPosition="below" size="large">Large Item</Item>
+				<Item centered inline label="label above" labelPosition="above" size="large">Large Item</Item>
+				<Item centered inline label="label before" labelPosition="before" size="large">Large Item</Item>
+				<Item centered inline label="label after" labelPosition="after" size="large">Large Item</Item>
+				<Item centered inline label="label below" labelPosition="below" size="small">Small Item</Item>
+				<Item centered inline label="label above" labelPosition="above" size="small">Small Item</Item>
+				<Item centered inline label="label before" labelPosition="before" size="small">Small Item</Item>
+				<Item centered inline label="label after" labelPosition="after" size="small">Small Item</Item>
```
Comment (jeonghee27) [inline samples/sampler/stories/qa/Item.js:115]: There is no case for `centered` with `slotBefore` / `slotAfter`

**Comment by jeonghee27** [comment]
https://github.com/enactjs/agate/pull/547#issuecomment-804507210
I think the position of slotAfter is still wrong. 

https://github.com/enactjs/agate/pull/547#issuecomment-804527420

**Comment by jeonghee27** [comment]
image

The max-width is not 720, it's 700

**Comment by jeonghee27** [comment]
I keep talking about this. 
Am I misunderstanding the GUI?
As far as I understand, slotBefore after should be located at both ends.

image
image

**Comment by jeonghee27** [comment]
I am constantly verifying the implementation and GUI. Please review again if it matches the GUI.
 @daniel-stoian-lgp

**Review by daniel-stoian-lgp — APPROVED** [review]
LGTM

File: Item/Item.js [hunk 17]
```diff
@@ -20,11 +20,13 @@ import Slottable from '@enact/ui/Slottable';
 import PropTypes from 'prop-types';
 import compose from 'ramda/src/compose';
 
-import {Marquee, MarqueeController} from '../Marquee';
+import {MarqueeBase, MarqueeController, MarqueeDecorator} from '../Marquee';
 import Skinnable from '../Skinnable';
 
 import componentCss from './Item.module.less';
 
+const Marquee = MarqueeDecorator({className: componentCss.marquee}, MarqueeBase);
```
Comment (jeonghee27) [inline Item/Item.js:28]: Could you leave the reason why you change this line on the PR description?

Comment (adrian-cocoara-lgp) [reply Item/Item.js:28]: I added it

**Comment by jeonghee27** [comment]
image
When I move focus to marquee item, the width is changed
Please check it.
 Do you want solve this bug in other issue? if so, please make the jira issue and leave the number after discussing your member.

**Comment by adrian-cocoara-lgp** [comment]
> image
> When I move focus to marquee item, the width is changed
> (qa-sampler > kitchecn Sink
> Please check it.
> Do you want to solve this bug in other issue? if so, please make the jira issue and leave the number after discussing your member.

I pushed some changes, it should be fixed now. Could you take a look, please?

**Comment by jeonghee27** [comment]
> I pushed some changes, it should be fixed now. Could you take a look, please?
Good, it resolved on Silicon. but it reproduced on Gallium.

**Review by jeonghee27 — APPROVED** [review]
LGTM except for one small bug..
 Item has a much influence, and the modification is not small. So I will pass this issue to someone to merge.

**Comment by adrian-cocoara-lgp** [comment]
> LGTM except for one small bug..
> Item has a much influence, and the modification is not small. So I will pass this issue to someone to merge.

I fixed the centered inline layout bug. 
The inline layout bug on Gallium is reproducing also on develop, I will create a ticket and solve it separately since it doesn't have to do with Silicon styles.

File: CHANGELOG.md [hunk 18]
```diff
@@ -39,6 +39,7 @@ The following is a curated list of changes in the Enact agate module, newest cha
 - `agate/Button` style to match latest design for Silicon skin
 - `agate/ImageItem` width/height proportion when `orientation` is `horizontal`
 - `agate/IncrementSlider` to match latest design for Silicon skin
+- `agate/Item` to to match latest design for Silicon skin
```
Comment (MikyungKim) [inline CHANGELOG.md:42]:
```suggestion
- `agate/Item` to match the latest design for Silicon skin
```

File: Item/Item.module.less [hunk 19]
```diff
@@ -47,11 +47,13 @@
 		.agate-item-sizing();
 
 		line-height: @agate-item-line-height;
+		max-width: @agate-item-max-width;
```
Comment (MikyungKim) [inline Item/Item.module.less:50]: Could we remove this property? I think it's too restrict even it's on the guide.

File: Dropdown/Dropdown.module.less [hunk 20]
```diff
@@ -175,6 +175,8 @@
 				border-width: @agate-dropdown-item-border-width;
 				padding-left: @agate-dropdown-item-padding;
 				padding-right: @agate-dropdown-item-padding;
+				height: @agate-dropdown-item-height;
+				max-width: @agate-dropdown-item-max-width;
```
Comment (MikyungKim) [inline Dropdown/Dropdown.module.less:179]: I don't think it's needed.

File: Item/Item.js (hunk 17 above)
Comment (MikyungKim) [inline Item/Item.js:28]: I think you meant below... 🤔 `MarqueeBase` should not be wrapped here. This is why the unit tests were failed.

```suggestion
const Marquee = MarqueeDecorator({className: componentCss.marquee}, 'div');
```

File: Item/tests/Item-specs.js [hunk 21]
```diff
@@ -155,7 +155,7 @@ describe('Item Specs', () => {
 		const subject = mount(<ItemBase>Hello מצב תמונה</ItemBase>);
 
 		const expected = 'rtl';
-		const actual = subject.find('.text').prop('style');
+		const actual = subject.find('.text').at(1).prop('style');
```
Comment (MikyungKim) [inline Item/tests/Item-specs.js:158]: Please revert this.

File: Dropdown/tests/Dropdown-specs.js [hunk 22]
```diff
@@ -13,7 +13,7 @@ describe('Dropdown', () => {
 		);
 
 		const expected = title;
-		const actual = dropDown.find('.text').text();
+		const actual = dropDown.find('.text').at(1).text();
```
Comment (MikyungKim) [inline Dropdown/tests/Dropdown-specs.js:16]: Please revert these.

File: Item/Item.module.less [hunk 23]
```diff
@@ -60,10 +62,12 @@
 
 				&.labelBefore .label {
 					.padding-start-end(0, @agate-item-label-padding);
+					align-self: center;
```
Comment (MikyungKim) [inline Item/Item.module.less:65]: Oh.. we have `align` property for `Layout` component.
We could re-write this in Item.js. at line 132,
```
<Layout align={labelPosition === 'before' || labelPosition === 'after' ? "center" : null} orientation={orientation}>
```

File: Item/Item.module.less [hunk 24]
```diff
@@ -72,13 +76,11 @@
...
 			.labelBefore {
 				.content {
-					box-sizing: border-box;
 					min-width: @agate-item-content-width;
 				}
 
 				.label {
-					box-sizing: border-box;
-					max-width: @agate-item-label-max-width;
+					max-width: @agate-item-label-width;
```
Comment (MikyungKim) [inline Item/Item.module.less:83]: Why did you change this?

File: styles/variables-silicon.less [hunk 25]
```diff
@@ -213,19 +213,28 @@
...
 // ---------------------------------------
 @agate-item-border-width:     0 0 3px 0;
+@agate-item-centered-inline-max-width: 480px;
 @agate-item-font-size:        33px;
 @agate-item-font-weight:      normal;
 @agate-item-height:           102px;
-@agate-item-height-large:     120px;
 @agate-item-side-outset:      30px;
 @agate-item-sub-text-size:    24px;
+@agate-item-inline-content-max-width: 639px;
```
Comment (MikyungKim) [inline styles/variables-silicon.less:222]: Would you please align this with the inline one?

File: Item/Item.module.less [hunk 26]
```diff
@@ -105,43 +107,95 @@
...
+			min-width: @agate-item-inline-min-width;
 			max-width: @agate-item-inline-max-width;
 
 			.itemContent {
 				max-width: @agate-item-inline-content-max-width;
 			}
 
+			&:not(.centered) {
+				.itemContent {
+					min-width: @agate-item-inline-notcentered-itemcontent-min-width;
```
Comment (MikyungKim) [inline Item/Item.module.less:119]: Where did you find this specification?

File: styles/variables-silicon.less (hunk 25 above)
Comment (adrian-cocoara-lgp) [reply styles/variables-silicon.less:222]: Actually this line will be removed from Silicon since I see there is a bug in storybook with Item inline and long labelAfter. The width of Item will grow on focus

File: Item/Item.module.less [hunk 27]
```diff
@@ -105,43 +107,95 @@
...
 
 		&.centered {
+			// override Marquee's alignment.
+			.marquee {
+				text-align: center;
+			}
+
 			.itemContent {
-				max-width: fit-content;
+				max-width: 100%;
```
Comment (MikyungKim) [inline Item/Item.module.less:137]: Would you please why you had to change this? This made the marquee's width to be 100% so that this whole "override Marquee alignment" happened... ;(

File: Item/Item.module.less [hunk 28]
```diff
@@ -105,43 +107,95 @@
...
+		}
 
 		// TODO: needs to be modified when new requirements for small item is available
 		&.small {
+			height: @agate-item-small-height;
 			font-size: @agate-item-small-font-size;
 			line-height: @agate-item-small-line-height;
 
 			&.hasLabel {
+				height: @agate-item-small-haslabel-height;
```
Comment (MikyungKim) [inline Item/Item.module.less:197]: Is this needed?

File: Item/Item.module.less [hunk 29]
```diff
@@ -156,37 +210,45 @@
 			}
 		}
 
+		.disabled({
+			color: @agate-item-disabled-color;
+			border-color: @agate-item-disabled-border-color;
+			opacity: @agate-item-disabled-opacity;
+		});
```
Comment (MikyungKim) [inline Item/Item.module.less:217]: Would you please describe why did you add this?

File: SwitchItem/SwitchItem.module.less [hunk 30]
```diff
@@ -6,6 +6,13 @@
...
 	.label {
 		vertical-align: middle;
+		position: absolute;
+		right: 0;
+
+		.enact-locale-rtl({
+			right: initial;
+			left: 0;
+		});
 	}
```
Comment (MikyungKim) [inline SwitchItem/SwitchItem.module.less:16]: I think this class can be removed if we could make the before and after label in Item vertically center well.

File: Dropdown/Dropdown.module.less (hunk 20 above)
Comment (adrian-cocoara-lgp) [reply Dropdown/Dropdown.module.less:179]: The `max-width` property could be removed since the item doesn't have the max-width anymore, but we still need the `height` property because now small item (in dropdown's case the radioItem) has a height of 99px and the layout is broken as seen below
dropdown

File: Item/Item.js (hunk 17 above)
Comment (adrian-cocoara-lgp) [reply Item/Item.js:28]: I see, thanks! 👍

File: Item/Item.module.less (hunk 23 above)
Comment (adrian-cocoara-lgp) [reply Item/Item.module.less:65]: Good tip

File: Item/Item.module.less (hunk 29 above)
Comment (adrian-cocoara-lgp) [reply Item/Item.module.less:217]: I needed to overwrite the styles for `disabled` which comes from `ThemeDecorator`, in order for the Item to have the same opacity when `disabled`. 
GUI did not take into consideration the @agate-disabled-opacity: 0.6 which is applied in ThemeDecorator

File: Item/Item.module.less (hunk 24 above)
Comment (adrian-cocoara-lgp) [reply Item/Item.module.less:83]: My mistake, I will change it back to `@agate-item-label-max-width`

File: Item/Item.module.less (hunk 26 above)
Comment (adrian-cocoara-lgp) [reply Item/Item.module.less:119]: This variable needed to be created in order to fix some layout bug for .itemContent when it is not centered.
I found it in GUI specs:
item-width

File: Item/Item.module.less (hunk 27 above)
Comment (adrian-cocoara-lgp) [reply Item/Item.module.less:137]: I had to made this change because there is a new specification for centered Item as seen below:
centered-gui

If we keep `max-width: fit-content`, when Item is centered, the icons would not position at the start for `labelBefore` and at the end for `labelAfter`. This is a screenshot of centered Item with `max-width: fit-content`.
fit-content

File: Item/Item.module.less (hunk 28 above)
Comment (adrian-cocoara-lgp) [reply Item/Item.module.less:197]: If we remove this line, there would be a layout bug on Gallium skin when small Item has a label, as seen below:
small-height-gallium

We also need a variable for Silicon because the height of small item with label will grow over 99px if this line is removed

File: SwitchItem/SwitchItem.module.less (hunk 30 above)
Comment (adrian-cocoara-lgp) [reply SwitchItem/SwitchItem.module.less:16]: Yea, it seems this is not needed anymore because of the `Layout` alignment we did in Item.js.

File: Dropdown/Dropdown.module.less (hunk 20 above)
Comment (MikyungKim) [reply Dropdown/Dropdown.module.less]: I meant "max-width". 👌

File: Item/Item.module.less (hunk 26 above)
Comment (MikyungKim) [reply Item/Item.module.less:115]: 🤔 I think the guide shows the padding of the content, not the min-width. I don't think it's neccessary.

File: Item/Item.module.less (hunk 29 above)
Comment (MikyungKim) [reply Item/Item.module.less:213]: 🤔  I think it's okay to leave as it is.

File: Item/Item.js [hunk 31]
```diff
@@ -116,7 +118,7 @@ const ItemContent = kind({
 	},
 
 	// eslint-disable-next-line enact/prop-types
-	render: ({centered, content, css, marqueeOn, label, orientation, styler, ...rest}) => {
+	render: ({centered, content, css, marqueeOn, label, labelPosition, orientation, styler, ...rest}) => {
 		delete rest.labelPosition;
```
Comment (MikyungKim) [inline Item/Item.js:122]: Please remove this.

File: Item/Item.module.less (hunk 28 above)
Comment (MikyungKim) [reply Item/Item.module.less:193]: 🆗

File: Item/Item.module.less (hunk 27 above)
Comment (MikyungKim) [reply Item/Item.module.less:133]: I see. I made some changes based on yours while keeping `max-width: fit-content` here.
You can check it out from https://github.com/enactjs/agate/compare/feature/PLAT-137544...feature/PLAT-137544-mk?expand=1.
Please take a look and you're welcome to review/apply/merge on your branch.

File: Item/Item.module.less (hunk 26 above)
Comment (adrian-cocoara-lgp) [reply Item/Item.module.less:115]: This variable needs to be set in order to avoid a bug with inline Item. When there is a long content and a long label with position after, when hovering over the Item, the width of Item will grow.

File: Item/Item.module.less (hunk 27 above)
Comment (adrian-cocoara-lgp) [reply Item/Item.module.less:133]: Your solution works fine, thanks. Small adjustments had to be made to fix some bugs on Gallium skin for inline Item with slatBefore and slotAfter.

File: Item/Item.module.less (hunk 29 above)
Comment (adrian-cocoara-lgp) [reply Item/Item.module.less:213]: 👍

File: Item/tests/Item-specs.js [hunk 32]
```diff
@@ -56,31 +56,27 @@ describe('Item Specs', () => {
 	});
 
 	test('should support `slotBefore`', () => {
-		const expected = 'slot before';
-
 		const item = mount(
-			<Item slotBefore={expected}>
+			<Item slotBefore="slot before">
```
Comment (MikyungKim) [inline Item/tests/Item-specs.js:60]: why did you change this?

Comment (adrian-cocoara-lgp) [reply Item/tests/Item-specs.js:60]: Unit tests had to be adjusted in order to pass because of the change we did for `slotBefore` and `slotAfter` in computed block in `Item.js`

File: Item/Item.module.less [hunk 33]
```diff
@@ -105,43 +104,93 @@
...
+					}
+				}
+
+				&.labelAbove,
+				&.labelBelow {
+					.content {
+						max-width: 100%;
+					}
+
+					.label {
```
Comment (MikyungKim) [inline Item/Item.module.less:148]:
```suggestion
					.content, .label {
```

File: styles/colors-silicon-day.less [hunk 34]
```diff
@@ -206,13 +207,14 @@
 // ---------------------------------------
 @agate-item-border-color:                 fade(@agate-black, 40%);
 @agate-item-color:                        @agate-text-color;
-@agate-item-label-color:                  @agate-noble-gray;
-@agate-item-focus-bg-color:               fade(@agate-black, 15%);
+@agate-item-label-color:                  fade(@agate-foreground, 70%);
+@agate-item-label-focus-selected-color:   fade(@agate-foreground, 70%);
```
Comment (MikyungKim) [inline styles/colors-silicon-day.less:211]: 🤔 I think... we don't want to set color with opacity here. The guide should be fixed. Let's not change these.

File: Item/Item.module.less (hunk 26 above)
Comment (MikyungKim) [reply Item/Item.module.less:115]: Have you tried with the recent code? I think it's not reproduced without this.

File: Item/Item.module.less [hunk 35]
```diff
@@ -105,43 +104,93 @@
...
 			.content {
 				flex-shrink: 1;
-				overflow: auto;
-			}
-
-			&.hasLabel {
-				.slotAfter {
-					.padding-start-end(@agate-item-slotafter-padding-left, 0);
-				}
+				overflow: hidden;
```
Comment (MikyungKim) [inline Item/Item.module.less:122]: Is this needed change?

File: Item/Item.module.less (hunk 26 above)
Comment (MikyungKim) [reply Item/Item.module.less:115]: For gallium, should we have `min-width` for `inline` or not?

Comment (MikyungKim) [reply Item/Item.module.less:115]: I think the below code instead of line 104 to 124 works fine for &.inline block. please check out and modify for sillicon skin.

```
		&.inline {
			// overriding ui/Item's inline-block
			display: inline-flex;
			max-width: @agate-item-inline-content-max-width;

			.itemContent {
				min-width: @agate-item-inline-min-width;
			}

		}
```

Comment (adrian-cocoara-lgp) [reply Item/Item.module.less:115]: For Gallium skin, the inline style issue will be taken care of in PLAT-139507(https://jira2.lgsvl.com/browse/PLAT-139507).

File: Item/Item.module.less (hunk 35 above)
Comment (adrian-cocoara-lgp) [reply Item/Item.module.less:122]: I deleted it

File: Item/tests/Item-specs.js (hunk 32 above)
Comment (MikyungKim) [reply Item/tests/Item-specs.js:60]: I think we could use `last()` instead of `first()`.

**Review by MikyungKim — APPROVED** [review]
:justshipit:
